### PR TITLE
add spring security dependency and SecurityConfig

### PR DESCRIPTION
**Pull request by bhtibrewal**
Date: 19/2/24

Developer Name: @bhtibrewal 

---

## Issue Ticket Number
https://github.com/Real-Dev-Squad/skill-tree-backend/issues/89

## Description

Set CORS policy 

### Documentation Updated?

- [ ] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [ ] No

### Development Tested?

- [ ] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
NA: only CORS setup done

</details>

### CODE REVIEW

File: skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.List;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf(AbstractHttpConfigurer::disable)
```
Comment (bhtibrewal) [inline skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java:23]: Discussed with @ankushdharkar will be handled separately  https://github.com/Real-Dev-Squad/skill-tree-backend/issues/93

File: skill-tree/src/main/resources/application.properties [hunk 2]
```diff
@@ -8,4 +8,4 @@ jwt.rds.public.key=${RDS_PUBLIC_KEY}
 API_V1_PREFIX=/api/v1
 spring.datasource.version=8.1.0
 management.endpoints.web.exposure.include=health,info,metrics
-
+logging.level.root=DEBUG
```
Comment (iamitprakash) [inline skill-tree/src/main/resources/application.properties:11]: please change this for Error only for here, and debug you can use for local system..?

File: skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+
+import java.util.Arrays;
+import java.util.List;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+
+
```
Comment (satyam73) [inline skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java]: NIT: extra line spacing here

File: skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf(AbstractHttpConfigurer::disable)
+                .cors(httpSecurityCorsConfigurer -> httpSecurityCorsConfigurer.configurationSource(corsConfigurationSource()));
+         return http.build();
+    }
+    @Bean
```
Comment (satyam73) [inline skill-tree/src/main/java/com/RDS/skilltree/Config/SecurityConfig.java:27]: NIT: we can add one line spacing here

**Pull request merged**